packages/react/src/index.ts

```diff
@@ -83,6 +87,63 @@ export function useEthSSOModal() {
     // TODO: Timeout at some point?
   }
 
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async function transaction(options: {
+    chainId: string;
+    transaction: string;
+  }): Promise<void> {
+    if (!ethSSOModalGlobal) {
+      throw new Error(
+        'Please call "createWeb3Modal" before using "useWeb3Modal" hook',
+      );
+    }
+    ethSSOModalGlobal.open();
+    const redirectUrl = ethSSOModalGlobal.options.redirectUrl;
+    ModalController.events.addEventListener("providerSelected", (evt) => {
```

Comment:
maybe better if dapp provides selectedProvider (or we default to the one used for authentication)

Reply:
changed a logic for a bit if is no provider set will throw a error